docs/blog/blog.md

```diff
@@ -0,0 +1,221 @@
+
+### Hashing
+
+This encoder uses the hashing trick to represent a high dimension space in a low dimension space while preserving sparsity. [[2](https://booking.ai/dont-be-tricked-by-the-hashing-trick-192a6aae3087),[ 3](https://www.youtube.com/watch?v=SLqKepl9rEo&t=473s)]
+
+- Process
+  - Converts variable to string “variable=category”, e.g., “modelname=Intel”
+  - Converts this string to an integer using a hashing function
+  - Modulus this integer by the length of output vector required
+  - Uses that index for indicator 1 (Essentially one-hot encoding in lower dimension)
+- Pros
+  - Fast and saves memory as the output vector is in a smaller dimension
+  - Can be used for multiple categorical variables at the same time
+- Cons
+  - Collisions happen if the output vector size is small
+  - Inverse mapping is not available so no interpretation
+
+### Word Based Encoders
+
+#### Language Model: Fasttext Encoder [[4](https://arxiv.org/abs/1607.04606)]
+
+- Process
+  - Downloads a pre-trained language model on English words
+  - For each category, performs a forward run of the pre-trained model
+  - Outputs the vectors from the last layer of the model
+- Pros
+  - Dynamic length of the output vector
+  - Captures meaning in the words or categories
+- Cons
+  - Heavy file size of the pre-trained model
+  - May not work well with words that don't appear a lot in natural English
+  - Output vector dimensions have no explainability
+
+##### Similarity Encoder [[5](https://dirty-cat.github.io/stable/generated/dirty_cat.SimilarityEncoder.html)]
+
+- Process
+  - Create one column per category
+  - Define n-gram similarity
+  - For each category, compare similarity with other categories
+  - Instead of 0 and 1 in one-hot encoding use similarity value
+  - E.g. Categories: Associate Software Developer, Software Developer, Senior Software Developer
+- Pros
+  - In high cardinality, categorical variables different entries are often variations on the same entities
+  - Using simple one-hot encoding will create orthogonal features, whereas it is clear that those 3 terms have a lot in common.
+  - If we wanted to use word embedding methods such as word2vec, we would have to go through a cleaning phase: those algorithms are not trained to work on data such as ‘Accountant/Auditor I’. However, this can be error-prone and time-consuming.
+  - The high dimensions can be meaningfully reduced to lower dimensions: All to Software developers, similar to clustering.
+- Cons
+  - High dimensional output vector
+  - Works only with variables that have meaningful substring overlaps
+
+##### Online Gamma Poisson Factorization [[6](https://arxiv.org/pdf/1907.01860.pdf)]
+
+- Process
+  - Estimates a decomposition of the string entries in terms of a linear combination of latent categories.
+  - Similar to LDA for finding topics in documents but since the string entries here are much shorter than text documents and can contain typos, it uses ngrams level representation.
```

Comment:
```suggestion
  - Similar to [LDA](https://en.wikipedia.org/wiki/Latent_Dirichlet_allocation) for finding topics in documents but since the string entries here are much shorter than text documents and can contain typos, it uses ngrams level representation.
```